File: app-java/src/main/java/tech/thdev/java_udemy_sample/util/ActivityUtil.java
package tech.thdev.java_udemy_sample.util;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

/**
 * Created by tae-hwan on 10/5/16.
 */

public class ActivityUtil {

    /**
     * The {@code fragment} is replace to the container view with id {@code frameId}. The operation is
     * performed by the {@code fragmentManager}.
     **/
    public static void replaceFragmentToActivity(@NonNull FragmentManager fragmentManager,
                                                 @NonNull Fragment fragment, int frameId) {
        FragmentTransaction transaction = fragmentManager.beginTransaction();
        transaction.replace(frameId, fragment);
        transaction.commit();
    }


}

File: app-kotlin/src/main/java/tech/thdev/kotlin_udemy_sample/util/ActivityUtil.kt
package tech.thdev.kotlin_udemy_sample.util

import android.content.Context
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import tech.thdev.kotlin_udemy_sample.constant.Constant
import tech.thdev.kotlin_udemy_sample.data.RecentPhotoItem
import tech.thdev.kotlin_udemy_sample.view.detail.DetailMoreActivity


/**
 * Created by tae-hwan on 10/3/16.
 */

fun AppCompatActivity.replaceFragmentToActivity(fragment: Fragment, frameId: Int) {
    val transaction = this.supportFragmentManager.beginTransaction()
    transaction.replace(frameId, fragment)
    transaction.commit()
}

fun Context.createDetailIntent(list: ArrayList<RecentPhotoItem>, position: Int): Intent {
    return Intent(this, DetailMoreActivity::class.java).apply {
        putParcelableArrayListExtra(Constant.KEY_PHOTO_DATA, list)
        putExtra(Constant.KEY_SHOW_POSITION, position)
    }
}

fun Context.createDetailIntent(item: RecentPhotoItem) =
    createDetailIntent(arrayListOf(item), 0)